packages/components/src/HeaderBar/HeaderBar.component.js

```diff
@@ -179,27 +182,28 @@ User.propTypes = {
 	}),
 };
 
-function Products({ renderers, ...props }) {
+function PureProducts({ renderers, t, ...props }) {
 	return (
 		<li className={theme['tc-header-bar-action']}>
 			<renderers.ActionDropdown
 				bsStyle="link"
 				className={theme['tc-header-bar-products']}
 				icon="talend-launcher"
-				label="Apps"
+				label={t('APPS')}
 				pullRight
 				tooltipPlacement="bottom"
 				{...props}
 			/>
 		</li>
 	);
 }
-
-Products.propTypes = {
+PureProducts.propTypes = {
 	renderers: React.PropTypes.shape({
 		ActionDropdown: React.PropTypes.func,
 	}),
+	t: React.PropTypes.func.isRequired,
 };
+const Products = translate('tui-components-headerBar')(PureProducts);
```

Comment:
we should be able to only translate the higher components and spread t props in it